### DISCUSSION TITLE
Fcpx 11 doesnt work with command post

### DISCUSSION THREAD

**Discussion by logChun15** [CommandPost/CommandPost#3393] [Technical Support] [answered]
CommandPost Error : The supplied FCPXML did not pass DTD validation. Please check that the FCPXML supplied is valid and try again.

**Comment by latenitefilms** [comment, marked as answer]
This has been fixed in the GitHub repository, but we haven't pushed out an update yet, as we're trying to fix some other things introduced in Final Cut Pro 11. Hopefully we'll have an update out tomorrow if all goes to plan.

**Reply by hokhim** [reply]
hi, how's the plan update running ?

**Reply by jcmourey** [reply]
I'm using FinalCutPro 11.1. Having a lot of difficulty. Is there still a compatibility issue with the current release?

**Reply by latenitefilms** [reply]
@jcmourey - What issues are you having exactly? CommandPost 1.5.1 should be working fine with Final Cut Pro 11.1.

**Reply by jcmourey** [reply]
I have a series of compound clips in my library named with the syntax "sXXX TOP" where XXX is a 3-digit number, like s005 TOP, s011 TOP, s149 TOP, etc.
I also have a list of video clips named "sXXX SUBTITLE TRACK" in my library.
I would like to write a LUA script in CommandPost to automatically add the "sXXX SUBTITLE TRACK" to each corresponding "sXXX TOP" compound clip, above all other clips in the compound clip's timeline, at timecode 00:00:00 of the compound clip.
I would also like to apply a pre-programmed GreenScreen effect to that "sXXX SUBTITLE TRACK" clip once added to the compound clip (the effect can be copied from a BASE SUBTITLE TRACK clip present in a BASE TIMELINE compound clip).

Bottom line is: is it possible to do that in CommandPost ? I'm an expert programmer, but a beginner in CommandPost and LUA.

To test my installation, I tried running some of the sample action snippet. This one works:

```
-----------------------------------------
-- Select all even clips in Timeline
-----------------------------------------
cp.apple.finalcutpro:launch()
local clips = cp.apple.finalcutpro:timeline():contents():clipsUI()
table.sort(clips, function(a, b) return a.AXPosition.x < b.AXPosition.x end)
local evenClips = {}
for i, clip in ipairs(clips) do
    if (i % 2 == 0) then
        table.insert(evenClips, clip)
    end
end
cp.apple.finalcutpro:timeline():contents():selectClips(evenClips)
```

But this doesn't work:

```
local fcp = require "cp.apple.finalcutpro"
fcp.preferencesWindow:show()
```
I get this error:
```
...ensions/cp/apple/finalcutpro/prefs/PreferencesWindow.lua:112: attempt to call a nil value (method 'mainMenuName')
stack traceback:
	...ensions/cp/apple/finalcutpro/prefs/PreferencesWindow.lua:112: in method 'show'
	[string "local fcp = require "cp.apple.finalcutpro"..."]:2: in main chunk
	[C]: in function 'xpcall'
	...armProjects/CommandPost/src/extensions/hs/_coresetup.lua:546: in function <...armProjects/CommandPost/src/extensions/hs/_coresetup.lua:522>
```

I tried this:

```
local allClips = cp.apple.finalcutpro.main.LibrariesBrowser:clips(function(clip)
    return true
end) or {}
```
Getting this error:
```
[string "..."]:2: attempt to index a nil value (field 'main')
stack traceback:
	[string "..."]:2: in main chunk
	[C]: in function 'xpcall'
	...armProjects/CommandPost/src/extensions/hs/_coresetup.lua:546: in function <...armProjects/CommandPost/src/extensions/hs/_coresetup.lua:522>
```

and this:
```

local browser = require("cp.apple.finalcutpro.main.LibrariesBrowser")
-- Use the 'clips' method with a filter function that returns true for every clip.
local allClips = browser:clips(function(clip)
    return true
end) or {}
```
Getting this error:
```
...xtensions/cp/apple/finalcutpro/main/LibrariesBrowser.lua:388: attempt to call a nil value (method 'isListView')
stack traceback:
	...xtensions/cp/apple/finalcutpro/main/LibrariesBrowser.lua:388: in method 'clips'
	[string "local browser = require("cp.apple.finalcutpro..."]:3: in main chunk
	[C]: in function 'xpcall'
	...armProjects/CommandPost/src/extensions/hs/_coresetup.lua:546: in function <...armProjects/CommandPost/src/extensions/hs/_coresetup.lua:522>
```

**Reply by jcmourey** [reply]
No replies or should I post this somewhere else?

**Reply by latenitefilms** [reply]
Apologies for the delayed reply!

I've create a new issue here:

https://github.com/CommandPost/CommandPost/issues/3433

I'll try and have a proper look at this soon.

**Comment by jcmourey** [comment]
Thanks!On 9 Apr 2025, at 02:48, Chris Hocking ***@***.***> wrote:﻿
Apologies for the delayed reply!
I've create a new issue here:
#3433
I'll try and have a proper look at this soon.
